### PR TITLE
cfg-parser: fix flag buffer overflow

### PR DESCRIPTION
**Pull request by OverOrion**
If the input was at least 32 bytes long, then the terminating NULL character was placed on the flag[32] position instead of flag[31].

I was only able to reproduce it using [axosyslog](https://github.com/axoflow/axosyslog-docker/pkgs/container/axosyslog) images.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/cfg-parser.c [hunk 1]
```diff
@@ -462,17 +462,14 @@ cfg_parser_cleanup(CfgParser *self, gpointer instance)
 gboolean
 cfg_process_flag(CfgFlagHandler *handlers, gpointer base, const gchar *flag_)
 {
-  gint h;
-  gchar flag[32];
+  gsize h;
 
-  for (h = 0; flag_[h] && h < sizeof(flag); h++)
+  GString *flag = g_string_new(flag_);
```
Comment (alltilla) [inline lib/cfg-parser.c:467]: Make sure to use flag->str in the strcmp below.

Also please free up the new GString.

**Review by alltilla — COMMENTED** [review]
Nice catch!

File: news/bugfix-4689.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+    cfg-parser: fix buffer overflow in case flag was long
```
Comment (alltilla) [inline news/bugfix-4689.md:1]: I think this entry is not useful for users. Instead we should mention that we fixed a crash which occurred for long strings passed to the scope() option.

**Comment by kira-syslogng** [comment]
Build FAILURE

**Comment by kira-syslogng** [comment]
Build FAILURE

File: lib/cfg-parser.c (hunk 1 above)
Comment (OverOrion) [reply lib/cfg-parser.c:468]: Thanks!

File: news/bugfix-4689.md (hunk 2 above)
Comment (OverOrion) [reply news/bugfix-4689.md]: Yeah, reworded it, thanks!

**Comment by OverOrion** [comment]
@kira-syslogng retest this please;

File: lib/cfg-parser.c [hunk 3]
```diff
@@ -462,23 +462,20 @@ cfg_parser_cleanup(CfgParser *self, gpointer instance)
...
+  for (h = 0; flag->str[h] && h < flag->len; h++)
     {
-      if (flag_[h] == '_')
-        flag[h] = '-';
-      else
-        flag[h] = flag_[h];
+      if (flag->str[h] == '_')
+        flag->str[h] = '-';
     }
-  flag[h] = 0;
```
Comment (bazsi) [inline lib/cfg-parser.c:475]: we could also use normalize_flag() here.

**Review by bazsi — APPROVED** [review]

File: lib/cfg-parser.c (hunk 3 above)
Comment (OverOrion) [reply lib/cfg-parser.c:475]: Thanks, with that the function becomes much simpler!

File: news/bugfix-4689.md [hunk 4]
```diff
@@ -0,0 +1 @@
+    cfg-parser: fix crash in case a long string was passed to the `scope()` function.
```
[the remark this answers is not in the document]
Comment (OverOrion) [reply news/bugfix-4689.md]: Thank you, indeed this seems much more informative for the end user!

**Pull request merged**